src/cloudformation_cli_python_lib/utils.py

```diff
@@ -60,6 +60,14 @@ def deserialize(cls, json_data: MutableMapping[str, Any]) -> "RequestData":
                 setattr(req_data, key, Credentials(**creds))
         return req_data
 
+    def serialize(self) -> Mapping[str, Any]:
+        ser = self.__dict__
+        for key in ser:
+            if not key.endswith("Credentials") or ser[key] is None:
+                continue
+            ser[key] = getattr(self, key).__dict__
+        return ser
```

Comment:
thinking about it more, this could be a dict comprehension that loops over `self.__dict__.items()` and filters and does getattr as appropriate to definitely create a new dict.

Reply:
not sure how to decide when getattr is appropriate inside the dict comprehension, how about something like:

```
    def serialize(self) -> Mapping[str, Any]:
        ser = self.__dict__
        creds = {k: v for k, v in ser.items() if k.endswith("Credentials") and v}
        for key in creds:
            ser[key] = getattr(self, key).__dict__
        return ser
```

Reply:
sorry, i think i was being a bit slow and misunderstanding. the weird handling of the credential keys is to (de)serialize them, i.e. introduce a different behaviour if it is a credentials class? so this might work?

```python
return {
    key: value.__dict__ if key.endswith("Credentials") else value
    for key, value in self.__dict__.items()
    if value is not None
}
```

this would also filter any `None` values though, but i think the java ser/de does this, too. might be good to copy the credentials' dict, too (`value.__dict__.copy()`)